plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-products-v2-controller.php

```diff
@@ -58,6 +61,55 @@ class WC_REST_Products_V2_Controller extends WC_REST_CRUD_Controller {
 	 */
 	public function __construct() {
 		add_action( "woocommerce_rest_insert_{$this->post_type}_object", array( $this, 'clear_transients' ) );
+		$this->initialize_rest_api_cache();
+	}
+
+	/**
+	 * Get the default entity type for response caching.
+	 *
+	 * @return string|null The entity type.
+	 */
+	protected function get_default_response_entity_type(): ?string {
+		return 'product';
+	}
+
+	/**
+	 * Get data for ETag generation, excluding fields that change on each request.
+	 *
+	 * @param array           $data        Response data.
+	 * @param WP_REST_Request $request     The request object.
+	 * @param string|null     $endpoint_id Optional endpoint identifier.
+	 * @return array Cleaned data for ETag generation.
+	 *
+	 * @phpstan-ignore-next-line missingType.generics -- WP_REST_Request generic type not specified, not practical for WP REST API.
```

Comment:
This is actually coming from php-stubs, which is setting a generic type. I [created a fix upstream](https://github.com/php-stubs/wordpress-stubs/issues/408), but I think that will be released with next WordPress release.

In the meantime, setting the parameter to a default value should fix this error. Ideally, the value in the PR should fix this.

```suggestion
	 * @phpstan-param WP_REST_Request<array<string, mixed>> $request
```